Upstream-Status:Inappropriate [debian patch]

From: Niko Tyni <ntyni@debian.org>
Subject: Save local versions of CPANPLUS::Config::System into /etc/perl.
    
This is a configuration file and needs to go in /etc by policy.
Besides, /usr may not even be writable.
    
This mirrors the Debian setup of CPAN.pm in debian/cpan_config_path.

See #533707.

---
 cpan/CPANPLUS/lib/CPANPLUS/Configure.pm           |    1 +
 cpan/CPANPLUS/lib/CPANPLUS/Internals/Constants.pm |    3 +++
 2 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/cpan/CPANPLUS/lib/CPANPLUS/Configure.pm b/cpan/CPANPLUS/lib/CPANPLUS/Configure.pm
index baac91d..a3794de 100644
--- a/cpan/CPANPLUS/lib/CPANPLUS/Configure.pm
+++ b/cpan/CPANPLUS/lib/CPANPLUS/Configure.pm
@@ -276,6 +276,7 @@ Saves the configuration to the package name you provided.
 If this package is not C<CPANPLUS::Config::System>, it will
 be saved in your C<.cpanplus> directory, otherwise it will
 be attempted to be saved in the system wide directory.
+(On Debian systems, this system wide directory is /etc/perl.)
 
 If no argument is provided, it will default to your personal
 config.
diff --git a/cpan/CPANPLUS/lib/CPANPLUS/Internals/Constants.pm b/cpan/CPANPLUS/lib/CPANPLUS/Internals/Constants.pm
index 1a38200..6ee0d82 100644
--- a/cpan/CPANPLUS/lib/CPANPLUS/Internals/Constants.pm
+++ b/cpan/CPANPLUS/lib/CPANPLUS/Internals/Constants.pm
@@ -199,6 +199,9 @@ use constant CONFIG_USER_FILE    => sub {
                                     ) . '.pm';
                                 };
 use constant CONFIG_SYSTEM_FILE  => sub {
+                                    # Debian-specific shortcut
+                                    return '/etc/perl/CPANPLUS/Config/System.pm';
+                                    
                                     require CPANPLUS::Internals;
                                     require File::Basename;
                                     my $dir = File::Basename::dirname(
-- 
tg: (a508b62..) debian/cpanplus_config_path (depends on: upstream)
